File: app/src/main/java/com/example/mytemplate/main/view/activity/base/BaseActivity.java
package com.example.mytemplate.main.view.activity.base;

import android.annotation.SuppressLint;
import android.os.StrictMode;

import androidx.appcompat.app.AppCompatActivity;

@SuppressLint("Registered")
public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onStart() {
        super.onStart();

        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().permitAll().build());
    }
}

File: app/src/main/java/com/example/mytemplate/base/BaseActivity.kt
package com.example.mytemplate.base

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.StrictMode
import android.view.LayoutInflater
import androidx.annotation.IdRes
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.example.mytemplate.widget.CustomProgressDialog

@SuppressLint("Registered")
abstract class BaseActivity<T: ViewBinding>(private val bindingFactory: (LayoutInflater) -> T) : AppCompatActivity() {

    val progressDialog: CustomProgressDialog by lazy {
        CustomProgressDialog(this)
    }

    lateinit var binding: T

    override fun onStart() {
        super.onStart()

        StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().permitAll().build())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = bindingFactory(layoutInflater)
        setContentView(binding.root)
        configureToolbar()
        onViewCreated(savedInstanceState)
    }

    @IdRes
    abstract fun toolbarId(): Int?

    abstract fun onViewCreated(savedInstanceState: Bundle?)

    private fun configureToolbar(){
        toolbarId()?.let {
            setSupportActionBar(findViewById(it))
            supportActionBar?.setDisplayHomeAsUpEnabled(false)
        }
    }
}